src/resolvers/Mutation/weeklyReport.js

```diff
@@ -0,0 +1,32 @@
+const { getUserId } = require('../../utils');
+
+const weeklyReport = {
+
+  createWeeklyReport: async (parent, args, ctx, info) => {
+    const userId = getUserId(ctx);
+
+    const hasPermissions = ctx.request.user.roles.some(role =>
```

Comment:
In my opinion (you don't have to fix this! ):
Maybe we can move `hasPermissions` function to `utils.js` to reuse it

Reply:
I'll fix this later if our project is growing up. Thanks for suggesting!